README.md

```diff
@@ -24,3 +24,7 @@ Public Key: base64
 Signature: base64  
 Hash: base64  
 Timestamp: uint64 representation of the Unix timestamp (seconds since January 1st, 1970)  
+
+## Concatenation for hashing
+When concatenating strings for hashing, `"` and `\` characters must be escaped by adding a `\` characters before them. `"` becomes `\"` and `\` becomes `\\`. Then each string is put between two `"` and added in a JSON array using compact representation.  
+For example, <code>a<sub>1</sub>||a<sub>2</sub>||...||a<sub>n</sub></code> = <code>["esc(a<sub>1</sub>)","esc(a<sub>2</sub>)",...,"esc(a<sub>n</sub>)"]</code> where <code>a<sub>1</sub>, ..., a<sub>n</sub></code> are strings and `esc` is an escape function.
```

Comment:
I think the goal of this PR is to have something as clear as possible for hashing data. Here we have "esc is an escape function", if we don't know what characters to escape and by what we won't have the same results for the input string. I think it would be useful to specify what escape function we want to use

Reply:
The escape function replace `"` and `\`  by `\"` and `\\` respectively.
But I think it could be better to simply use a Json for the "array of byte" to hash...
Indeed, the Json format is standard whereas what we are doing here is prone to errors as we create the Json array by hand.

```suggestion
For example, `{1:"firstString",2:"secondString",...}` 
```

Reply:
Why do you want to use a JSON object instead of an array. What is the advantage?
Also, in your example you still need to escape strings, otherwise you could have problems. By example, `Hash(string1||string2",3:"string3) = Hash({1:"string1",2:"string2",3:"string3"}) = Hash(string1||string2||string3)`

Reply:
You're right, it's not a byte array so we definitely need an escape function.
I'm OK with this now, thanks

Reply:
okay, so I just don't understand why we want to escape characters that are not present in b64. I mean the strings that are going to be in this are base64-encoded right ?

Reply:
okay I thought we used the base64 encoded version but obviously not, so it's alright for me